### ISSUE TITLE
Certain Non-Simple Polygons Aren't Simplified

### ISSUE THREAD

**Issue by KermMartian** [Geopipe/PolygonSimplification#1]
For a number of polygons, the original computation of the NEF polyhedron for the non-simple shape's interior (the first line below), does not work; the second line was therefore necessary. However, that second line doesn't work for some shapes that worked with the first.
```
       NefPolyhedron area_pointset = (whole_plane - boundary_pointset).interior();
       NefPolyhedron area_pointset = (whole_plane - boundary_pointset).closure().interior();
```

This shape works with `.interior()`, but not `.closure().interior()`.
`(0, 4), (2, 1), (1, 1), (3, 4), (2, 4), (0, 0), (3, 0), (1, 4)`
![image](https://user-images.githubusercontent.com/1566659/48249468-3e6c0b80-e3c9-11e8-93d0-2e4ee158ff31.png)

This shape works with `.closure().interior()`, but not `.interior()`.
`(68.008225, 117.633952), (66.679413, 114.979042), (61.679573, 104.989577), (58.811962, 99.260213), (49.688695, 81.032319), (49.522233, 80.699734), (39.963931, 61.602657), (47.148790, 58.006555), (53.377595, 54.888968), (76.336922, 43.397566), (98.052066, 86.783490), (68.863934, 101.392479), (68.856716, 101.378057), (68.587088, 100.839352), (65.994932, 95.660336), (65.994669, 95.659811), (65.993593, 95.657661), (65.993290, 95.657056), (65.992888, 95.656252), (65.992886, 95.656248), (65.992867, 95.656211), (65.992805, 95.656086), (65.992619, 95.655716), (65.992585, 95.655647), (65.989494, 95.649471), (56.875023, 77.439151), (56.875014, 77.439132), (56.874488, 77.438081), (56.874402, 77.437910), (56.874376, 77.437858), (56.873492, 77.436091), (56.873282, 77.435672), (56.707092, 77.103631), (56.873554, 77.436216), (65.996821, 95.664111), (68.864432, 101.393474), (104.603296, 83.505770), (106.928331, 82.342063), (111.928171, 92.331529), (109.603136, 93.495235), (103.052404, 96.773951), (75.621855, 110.503249), (76.950668, 113.158159), (75.193084, 114.037850)`
![image](https://user-images.githubusercontent.com/1566659/48249765-0e713800-e3ca-11e8-96b6-30adb2795c95.png)

**Comment by elfprince13** [comment]
So, the first line is definitely correct, the question is what's going on later that's failing to clean up the latter shape?

**Comment by elfprince13** [comment]
Working notes: output from both, with the un-modified line in question:

```
[thomas@Mithlond] PolygonSimplification $ ./polygon-simplification 0 4 2 1 1 1 3 4 2 4 0 0 3 0 1 4
There shouldn't be a hole in the interior of this face, because it's the result of tracing a single polyline into the plane

Finished set:
	3 1 1 2 1 1.5 1.75  ( is simple? 1 )
	9 0 0 3 0 1.85714 2.28571 3 4 2 4 1.5 3 1 4 0 4 1.14286 2.28571  ( is simple? 1 )
[thomas@Mithlond] PolygonSimplification $ ./polygon-simplification 68.008225 117.633952 66.679413 114.979042 61.679573 104.989577 58.811962 99.260213 49.688695 81.032319 49.522233 80.699734 39.963931 61.602657 47.148790 58.006555 53.377595 54.888968 76.336922 43.397566 98.052066 86.783490 68.863934 101.392479 68.856716 101.378057 68.587088 100.839352 65.994932 95.660336 65.994669 95.659811 65.993593 95.657661 65.993290 95.657056 65.992888 95.656252 65.992886 95.656248 65.992867 95.656211 65.992805 95.656086 65.992619 95.655716 65.992585 95.655647 65.989494 95.649471 56.875023 77.439151 56.875014 77.439132 56.874488 77.438081 56.874402 77.437910 56.874376 77.437858 56.873492 77.436091 56.873282 77.435672 56.707092 77.103631 56.873554 77.436216 65.996821 95.664111 68.864432 101.393474 104.603296 83.505770 106.928331 82.342063 111.928171 92.331529 109.603136 93.495235 103.052404 96.773951 75.621855 110.503249 76.950668 113.158159 75.193084 114.037850
Finished set:
	52 39.9639 61.6027 47.1488 58.0066 53.3776 54.889 76.3369 43.3976 98.0521 86.7835 68.8639 101.392 68.8567 101.378 68.5871 100.839 65.9949 95.6603 65.9947 95.6598 65.9936 95.6577 65.9933 95.6571 65.9929 95.6563 65.9929 95.6562 65.9929 95.6562 65.9928 95.6561 65.9927 95.6558 65.9926 95.6557 65.9926 95.6556 65.9895 95.6495 56.8862 77.4616 56.875 77.4392 56.875 77.4391 56.8745 77.4381 56.8744 77.4379 56.8744 77.4379 56.8735 77.4361 56.8733 77.4357 56.7071 77.1036 56.8736 77.4362 56.875 77.4392 56.875 77.4392 56.8862 77.4616 65.9926 95.6557 65.9926 95.6557 65.9927 95.6558 65.9968 95.6641 68.8644 101.393 104.603 83.5058 106.928 82.3421 111.928 92.3315 109.603 93.4952 103.052 96.774 75.6219 110.503 76.9507 113.158 75.1931 114.038 68.0082 117.634 66.6794 114.979 61.6796 104.99 58.812 99.2602 49.6887 81.0323 49.5222 80.6997  ( is simple? 0 )
	9 56.7071 77.1036 56.8733 77.4357 56.8735 77.4361 56.8744 77.4379 56.8744 77.4379 56.8745 77.4381 56.875 77.4391 56.875 77.4392 56.8736 77.4362  ( is simple? 1 )
	3 56.875 77.4392 56.8862 77.4616 56.875 77.4392  ( is simple? 1 )
	4 56.8862 77.4616 65.9895 95.6495 65.9926 95.6556 65.9926 95.6557  ( is simple? 1 )
	3 65.9926 95.6557 65.9927 95.6558 65.9926 95.6557  ( is simple? 1 )
```

The correct output for the first polygon should probably not also output the overlapping inner triangle.

Working on distilling an MWE for the second example, but I'm reasonably sure it should also be a single shape.

**Comment by elfprince13** [comment]
Closed as of 4603ef9d6bac612f210985dc9c3d52f1f9db398f:
```
[thomas@Mithlond] PolygonSimplification $ ./polygon-simplification 0 4 2 1 1 1 3 4 2 4 0 0 3 0 1 4
Finished set:
	9 0 0 3 0 1.85714 2.28571 3 4 2 4 1.5 3 1 4 0 4 1.14286 2.28571  ( is simple? 1 )
[thomas@Mithlond] PolygonSimplification $ ./polygon-simplification 68.008225 117.633952 66.679413 114.979042 61.679573 104.989577 58.811962 99.260213 49.688695 81.032319 49.522233 80.699734 39.963931 61.602657 47.148790 58.006555 53.377595 54.888968 76.336922 43.397566 98.052066 86.783490 68.863934 101.392479 68.856716 101.378057 68.587088 100.839352 65.994932 95.660336 65.994669 95.659811 65.993593 95.657661 65.993290 95.657056 65.992888 95.656252 65.992886 95.656248 65.992867 95.656211 65.992805 95.656086 65.992619 95.655716 65.992585 95.655647 65.989494 95.649471 56.875023 77.439151 56.875014 77.439132 56.874488 77.438081 56.874402 77.437910 56.874376 77.437858 56.873492 77.436091 56.873282 77.435672 56.707092 77.103631 56.873554 77.436216 65.996821 95.664111 68.864432 101.393474 104.603296 83.505770 106.928331 82.342063 111.928171 92.331529 109.603136 93.495235 103.052404 96.773951 75.621855 110.503249 76.950668 113.158159 75.193084 114.037850
Finished set:
	33 39.9639 61.6027 47.1488 58.0066 53.3776 54.889 76.3369 43.3976 98.0521 86.7835 68.8639 101.392 68.8567 101.378 68.5871 100.839 65.9949 95.6603 65.9947 95.6598 65.9936 95.6577 65.9933 95.6571 65.9929 95.6563 65.9929 95.6562 65.9929 95.6562 65.9928 95.6561 65.9927 95.6558 65.9968 95.6641 68.8644 101.393 104.603 83.5058 106.928 82.3421 111.928 92.3315 109.603 93.4952 103.052 96.774 75.6219 110.503 76.9507 113.158 75.1931 114.038 68.0082 117.634 66.6794 114.979 61.6796 104.99 58.812 99.2602 49.6887 81.0323 49.5222 80.6997  ( is simple? 1 )
[thomas@Mithlond] PolygonSimplification $
```
